File: Programmers/연습문제/Level 1/콜라츠 추측/Solution.js
function solution(num) {
    var answer = getCollatz(num);
    return answer;
}

function getCollatz(num) {
    let n = num;
    let count = 0;
    while (n != 1 && count < 500) {
        if (n % 2) {
            n = n*3+1;
        } else {
            n /= 2;
        }
        count++;
    }
    return count >= 500 ? -1 : count;
}

File: Programmers/연습문제/Level 1/콜라츠 추측/Solution.swift
func solution(_ num:Int) -> Int {
    return isCollatz(num)
}

func isCollatz(_ num: Int) -> Int {
    var count = 0
    let maxCount = 500
    var _num = num
    
    while count < maxCount && _num != 1{
        _num = doCollatz(_num)
        count += 1
    }
    
    return count < maxCount ? count : -1
}

func doCollatz(_ num: Int) -> Int{
    if num.isMultiple(of: 2) {
        return num / 2
    } else {
        return num * 3 + 1
    }
}


// TC
let num = 6
print(solution(num))
